crates/ruma-events/src/room/join_rules.rs

```diff
@@ -155,6 +155,10 @@ impl Restricted {
     }
 }
 
+// FIXME: Any entries in the list which do not match the expected format are ignored. Thus, if all
+// entries are invalid, the list behaves as if empty and all users without an invite are rejected.
+// i.e. ser/de needs to be lax
```

Comment:
What is this fixme about? How could entries in the list be honored if they don't follow the expected format?

Reply:
The two sentences after `FIXME` are taken directly from the spec :shrug: 

https://github.com/matrix-org/matrix-doc/blob/main/proposals/3083-restricted-rooms.md#proposal Third actual "paragraph" or first one after bullet points.

Reply:
I'm not sure that's what that paragraph says. Either way our deserialization is already "lax" because we have `AllowRule::_Custom`, no?

Reply:
Oh are they more talking about the shape of the JSON, like it has to look like
```json
{
    "type": "m.some_supported_type",
    "room_id": "!realroomid"
}
```
or something they come up with in the future and __not__ that the individual pieces could be invalid.